File: frontend/src/services/estimate.ts
// src/services/estimate.ts
import api from "@/lib/axios";
import { EstimateRequest, EstimateResponse } from "@/models/estimate";

export const estimateService = {
  async estimateNutrition(input: EstimateRequest): Promise<EstimateResponse> {
    const { data } = await api.post<EstimateResponse>("/nutrition/estimate", input);
    return data;
  },
};

File: backend/app/routers/estimate.py
# app/routers/estimate.py
from fastapi import APIRouter
from app.schemas.estimate import EstimateRequest, EstimateResponse
from app.services.estimate import EstimateService

router = APIRouter(tags=["nutrition"])


@router.post("/nutrition/estimate", response_model=EstimateResponse)
async def estimate_nutrition(payload: EstimateRequest):
    return await EstimateService.estimate_nutrition(payload)
